flocker/control/httpapi.py

```diff
@@ -812,11 +818,15 @@ def replace_configuration(self, applications, deployment):
         :param deployment: Configuration of which applications run on
             which nodes.
         """
+        # XXX we're going to have to deal with UUIDs better,
+        # somehow... look up state of cluster? Change deployment
+        # configuration format?
         try:
             configuration = FigConfiguration(applications)
             if not configuration.is_valid_format():
                 configuration = FlockerConfiguration(applications)
             return self.persistence_service.save(model_from_configuration(
+                self.cluster_state_service.as_deployment(),
```

Comment:
No, I'm wrong, the agent *used* to get its local hostname from the command line.

And that seems still to be in the documentation for the zfs agent, but not for the container or dataset agents...
```
(1733)(UUID-in-API-FLOC-1733 ? =)[~/.../HybridLogic/flocker]$ flocker-zfs-agent --help
Usage: flocker-zfs-agent [OPTIONS] <local-hostname> <control-service-hostname>
```

Maybe update the help text for `flocker-zfs-agent`.